agent/config/config.go

```diff
@@ -373,20 +397,37 @@ func ec2MetadataConfig(ec2client ec2.EC2MetadataClient) Config {
 // error is returned, however, if the config is incomplete in some way that is
 // considered fatal.
 func NewConfig(ec2client ec2.EC2MetadataClient) (config *Config, err error) {
-	ctmp := environmentConfig() //Environment overrides all else
+	var errs []error
+	var errTmp error
+	ctmp, errTmp := environmentConfig() //Environment overrides all else
```

Comment:
`errTmp` is used in a few places in this function, not always associated with `environmentConfig()`. If you think using a unique identifier for each of them improves readability, I'm fine with that. I don't really agree with it, but don't have strong feelings either way.